react/utils/client/links/ensureSessionLink.ts

```diff
@@ -0,0 +1,36 @@
+import { ApolloLink, NextLink, Observable, Operation } from 'apollo-link'
+import { Subscription } from 'apollo-client/util/Observable'
+
+export const ensureSessionLink = (sessionPromise: Promise<any>) => {
+  const sessionRequiredForScope = new Set(['segment', 'private'])
```

Comment:
nitpick: this could be defined out of the function.

Reply:
It was explicit, since I don't want to generate unwanted objects in SSR. However it could harm our beloved moto g4 ... Maybe I'll change it